### PR TITLE
Api tests

### PR DESCRIPTION
**Pull request by adilenko**

### CODE REVIEW

**Comment by adilenko** [comment]
Infrostructue is created. But currently no time for extending test coverage.

**Comment by adilenko** [comment]
image

File: GL_Framework/libs/assertpy_extensions.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import jsonschema
+from assertpy import add_extension
+from json import loads
+from constants import SCHEMAS_DIRECTORY
+import allure
+
+
+@allure.step("Validate json schema")
+def schema_valid(self, schema_name: Optional[str], status_code: int = 200):
+    if self.val.status_code != status_code:
```
Comment (try-to-pick-unique-username) [inline GL_Framework/libs/assertpy_extensions.py:12]: The response status code verification should be moved to the API client or session class, and expected status code should not be hardcoded

File: GL_Framework/tests/api_test/test_login.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+from configuration.config import ConfigModel, get_config_variable_by_name
+from assertpy import assert_that, soft_assertions
+from helpers.fake import fake
+
+@pytest.mark.smoke
+def test_login():
+    resp = ApiClient.loggin(
+        get_config_variable_by_name(ConfigModel.user),
+        get_config_variable_by_name(ConfigModel.password))
+    assert_that(resp.status_code).is_equal_to(200)
```
Comment (try-to-pick-unique-username) [inline GL_Framework/tests/api_test/test_login.py:12]: Expected status codes should not be hardcoded, keep them in constants in configuration / data files

File: GL_Framework/tests/api_test/test_login.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+@pytest.mark.smoke
+def test_invalide_login():
+    name = fake.name()
+    password = fake.password()
+    resp = ApiClient.loggin(
+        name,
+        password)
+    with soft_assertions():
+        assert_that(resp.status_code).is_equal_to(401)
```
Comment (try-to-pick-unique-username) [inline GL_Framework/tests/api_test/test_login.py:22]: Expected status codes should not be hardcoded, keep them in constants in configuration / data files

File: GL_Framework/tests/api_test/test_login.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        assert_that(resp.text).is_equal_to("You are required to log in with a valid username and password")
+
+@pytest.mark.api
+@pytest.mark.parametrize("size", [58])
+def test_file_count(api_session, size):
+    resp = api_session.file_count()
+    with soft_assertions():
+        assert_that(resp.status_code).is_equal_to(200)
```
Comment (try-to-pick-unique-username) [inline GL_Framework/tests/api_test/test_login.py:30]: Expected status codes should not be hardcoded, keep them in constants in configuration / data files

File: GL_Framework/tests/api_test/test_login.py [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    password = fake.password()
+    resp = ApiClient.loggin(
+        name,
+        password)
+    with soft_assertions():
+        assert_that(resp.status_code).is_equal_to(401)
+        assert_that(resp.text).is_equal_to("You are required to log in with a valid username and password")
+
+@pytest.mark.api
+@pytest.mark.parametrize("size", [58])
```
Comment (try-to-pick-unique-username) [inline GL_Framework/tests/api_test/test_login.py:26]: Why 58? Parameter should be clear and kept in constants

**Review by try-to-pick-unique-username — COMMENTED** [review]
Please take a look on review comments
General comment - the following homework requirement is not met:
"Create 2-4 API tests for each endpoint"

**Pull request merged**